boa/__init__.py

```diff
@@ -45,17 +45,41 @@ def set_env(new_env):
     Env._singleton = new_env
 
 
+# Simple context manager which functions like the `open()` builtin -
+# if simply called, it never calls __exit__, but if used as a context manager,
+# it calls __exit__ at scope exit
+class _TmpEnvMgr:
+    def __init__(self, new_env):
+        global env
+        self.old_env = env
+
+        set_env(new_env)
+
+    def __enter__(self):
+        # dummy
+        pass
+
+    def __exit__(self, *args):
+        set_env(self.old_env)
```

Comment:
we probably need a test for this

Reply:
do you mean a more specific test? it is kind of tested already in fork tests (https://github.com/vyperlang/titanoboa/blob/7271bf54e744848f6a25d1568803b28fcd2e5f3c/tests/integration/fork/conftest.py)

Reply:
That doesn't check the exit functionality

Reply:
hmm i think it kind of does, otherwise there might be some kind of issues during fixture teardown. but sure, i can add a unit test.